File: JavaScript/RansomNote.js
const canConstruct = (ransomNote, magazine) => {
    const map = new Map()
    const ransomMap = new Map()
    let count = 0

    for (let i = 0; i < magazine.length; i++) {
        if (!map.has(magazine[i])) {
            map.set(magazine[i], 1)
        } else {
            map.set(magazine[i], map.get(magazine[i]) + 1)
        }
    }

    for (let j = 0; j < ransomNote.length; j++) {
        if (map.has(ransomNote[j]) && map.get(ransomNote[j]) > 0) {
            map.set(ransomNote[j], map.get(ransomNote[j]) - 1)
        } else {
            return false
        }
    }
    return true
};

File: Kotlin/RansomNote.kt
//8.28
class Solution {
    fun canConstruct(ransomNote: String, magazine: String): Boolean {
        val map = mutableMapOf<Char, Int>()

        for (i in magazine) {
            if (!map.containsKey(i)) {
                map.put(i, 1)
            } else {
                map.put(i, map.getValue(i) + 1)
            }
        }
        for (j in ransomNote) {
            if (map.containsKey(j) && map[j]!! > 0) {
                map.set(j, map[j]!! - 1)
            } else {
                return false
            }
        }
        return true
    }
}